improver/utilities/mathematical_operations.py

```diff
@@ -356,3 +356,57 @@ def process(self, cube):
         integrated_cube = self.perform_integration(upper_bounds_cube, lower_bounds_cube)
 
         return integrated_cube
+
+
+def fast_linear_fit(x_data, y_data, axis=None, keepdims=False, gradient_only=False):
+    """Uses a simple linear fit approach to calculate the
+    gradient along specified axis (default is to fit all points).
+    Uses vectorized operations, so it's much faster than using scipy lstsq
+    in a loop. This function does not handle NaNs, but will work with masked arrays.
+    Args:
+        x_data (numpy.ndarray):
+            x axis data.
+        y_data (numpy.ndarray):
+            y axis data.
+        axis (int or tuple of int):
+            Optional argument, specifies the axis to operate on.
+            Default is to flatten arrays and fit all points.
+        keepdims (bool):
+            If this is set to True, the axes which are reduced are left in the
+            result as dimensions with size one. With this option, the result
+            will broadcast correctly against the input array.
+        gradient_only (bool):
+            If true only returns the gradient.
+    Returns:
+        (tuple): tuple containing:
+            **gradient** (numpy.ndarray):
+                The gradient between x and y.
+            **y-intercept** (numpy.ndarray):
+                The calculated y-intercepts.
+    """
+    # Finds a compatible shape for the means to be reshaped into
+    if not x_data.shape == y_data.shape:
+        raise ValueError("Shape of x and y do not match")
+
+    # Check that there are no spurious NaNs in one of the arrays
+    # only - this will mess up the mean.
+    if not (y_data.mask == x_data.mask).all():
+        raise ValueError("Positions of NaNs do not match in x and y")
```

Comment:
This will fail with `AttributeError` for normal arrays.
```suggestion
    # Check that there are no mismatched masks (this will mess up the mean).
    if not (ma.getmask(y_data) == ma.getmask(x_data.mask)).all():
        raise ValueError("Mask of x and y do not match.")
```